parson.c

```diff
@@ -1516,42 +1719,45 @@ JSON_Status json_serialize_to_file_pretty(const JSON_Value *value, const char *f
     return return_code;
 }
 
-char * json_serialize_to_string_pretty(const JSON_Value *value) {
+char * json_serialize_to_string_pretty(const JSON_Value* value : itype(_Ptr<const JSON_Value>)) : itype(_Nt_array_ptr<char>) {
     JSON_Status serialization_result = JSONFailure;
     size_t buf_size_bytes = json_serialization_size_pretty(value);
-    char *buf = NULL;
+    _Array_ptr<char> buf : byte_count(buf_size_bytes) = NULL;
     if (buf_size_bytes == 0) {
         return NULL;
     }
-    buf = (char*)parson_malloc(buf_size_bytes);
+    buf = parson_malloc(char, buf_size_bytes);
```

Comment:
I think you should declare `buf` to have `_Nt_array_ptr` type and use `parson_string_malloc` here.